shapeworks_cloud/core/tasks.py

```diff
@@ -115,10 +121,16 @@ def run_shapeworks_command(
             with Popen(full_command, cwd=download_dir, stdout=PIPE, stderr=PIPE) as process:
                 if process.stderr and process.stdout:
                     for line in iter(process.stdout.readline, b''):
-                        if command == 'analyze':
-                            print(line.decode())  # analyze task has no xmloutput
+                        progress.refresh_from_db()
+                        if progress.abort:
```

Comment:
Yes, but for this particular executable we call, it gives regular updates for the progress output, so I doubt it would make a noticeable difference from polling every second. I wouldn't be opposed to the time-based poll, but I couldn't find an elegant way to accomplish it.

Reply:
I see, then if you're comfortable with it then carry on. Seeing the control of the process depend upon that process outputting something to stdout just stood out as potentially risky to me (without any context).

Reply:
If you wanted a bit more robust of a pattern for aborting while a subprocess is running, you could use the `select` syscall ([python has nice wrapping for this](https://docs.python.org/3/library/select.html)) with a fixed timeout in a loop, that way you wouldn't have to wait for stdout to kill the subprocess.

Reply:
Alternatively I think `epoll.from_fd` may be a more preferred (but perhaps less portable) mechanism.

Reply:
Thanks! I'll try that instead